Add Virtual Camera events and requests

TBGRenfold opened the pull request:
Based on:

https://github.com/Palakis/obs-websocket/blob/4.9.1/docs/generated/protocol.md#virtual-cam
https://github.com/Palakis/obs-websocket/blob/4.9.1/docs/generated/protocol.md#virtual-cam-1

TBGRenfold committed:
Add Virtual Camera events and requests.

Signed-off-by: TBG Renfold <tbg.renfold@gmail.com>

TBGRenfold committed:
Edit GetVirtualCamStatus return summary.

TBGRenfold committed:
Fix spelling mistakes in summaries.

TBGRenfold committed:
Add Callbak to delegate names.

Change the events from EventHandler to their respective callbacks.

TBGRenfold committed:
Remove callbacks in Events.cs (move back to EventHandlers) and question sanity of why I added them in the first place as they have no data to send to an event.

TBGRenfold committed:
Removal of using System from Events.cs

BarRaider commented on obs-websocket-dotnet/OBSWebsocket_Requests.cs:
```diff
@@ -53,7 +53,7 @@ public SourceScreenshotResponse TakeSourceScreenshot(string sourceName, string e
                 requestFields.Add("saveToFilePath", saveToFilePath);
             }
-            if (width > -1)
+            if (width > -1)   
```
please clean up

BarRaider commented on obs-websocket-dotnet/VirtualCamStatus.cs:
```diff
@@ -0,0 +1,29 @@
+        /// Builds the object from the JSON response body
+        /// </summary>
+        /// <param name="data">JSON response body as a <see cref="JObject"/></param>
+        public VirtualCamStatus(JObject data)
```
Please also add a default constructor (With no arguments), otherwise this will fail JSON deserialization.

BarRaider commented on obs-websocket-dotnet/OBSWebsocket_Requests.cs:
```diff
@@ -2028,5 +2028,40 @@ public void CreateScene(string sceneName)
+        /// <summary>
+        /// Toggle virtual camera on or off (depending on the current virtual camera state).
+        /// </summary>
+        public void StartStopVirtualCam()
```
Maybe rename to "ToggleVirtualCam" ?

BarRaider reviewed (changes requested):
Overall looks good, minor tweaks.

TBGRenfold committed:
Cleanup whitespace.

Rename StartStopVirtualCam to ToggleVirtualCam.

Fix spelling mistake in SendRequest in the ToggleVirtualCam method (oops missed that one).

Add default constructor to VirtualCamStatus.cs.

TBGRenfold committed:
Merge branch 'dev-4.9.1' into virtualcam

BarRaider commented:
Hey,
Was there a reason for commit 443facb "Merge dev-4.9.1 into Virtualcam" given this is being merged into dev-4.9.1?
Other than that it's ready to be merged but didn't understand why  that commit was needed.

TBGRenfold commented:
> Hey,
> Was there a reason for commit [443facb](https://github.com/BarRaider/obs-websocket-dotnet/commit/443facbb3507ac1fbe42185eb1d78159b296c23c) "Merge dev-4.9.1 into Virtualcam" given this is being merged into dev-4.9.1?
> Other than that it's ready to be merged but didn't understand why that commit was needed.

443facb << Shouldn't of happened. Bit of a PR noob, and realize that I should of made a seperate branch for merging 4.9.1 with any conflicts my end (testing against obs-ws).
